### PR TITLE
Added the ability for omdb to talk to crucible-agent

### PR DESCRIPTION
**Pull request by leftwo**
Made omdb a client of the crucible-agent
This allows for omdb to query a crucible agent for regions and snapshots.

```
root@BRM44220005:~# /tmp/omdb crucible-agent --crucible-agent-url http://\[fd00:1122:3344:105::c]:32345 regions list | grep -v destroyed
REGION_ID                            STATE     BLOCK_SIZE EXTENT_SIZE EXTENT_COUNT PORT  
0cb6e1dc-d8e5-48a3-9e60-20ff54202f28 created   512        131072      160          19004 
16e0ef4a-ed0e-49e0-b5ee-1a42be425356 created   512        131072      160          19002 
2a05cb22-2aff-4a7b-a308-96ada8618b84 created   512        131072      160          19001 
2ac4858f-291e-4ea3-aa07-33e966afabbb created   512        131072      32           19000 
753836fe-4f3d-415a-ba2b-87aa0e39d9cf created   512        131072      240          19044 
ba5840f6-7802-4926-bb5a-c4aa0999e850 created   512        131072      160          19003 
```

and
```
root@BRM42220009:~# /tmp/omdb crucible-agent --crucible-agent-url http://\[fd00:1122:3344:107::8\]:32345 regions list | grep -v destroyed
REGION_ID                            STATE     BLOCK_SIZE EXTENT_SIZE EXTENT_COUNT PORT  
0c827b69-72aa-4692-a0cd-00f20657092b created   512        131072      160          19004 
29739b45-e9d1-47d4-9f8b-e45e8b058c61 created   4096       16384       1280         19000 
524a6ad7-1ed9-49ab-ac8d-08d5cb866172 created   512        131072      16           19006 
7280049d-d02f-4bb5-af10-8452896b0dca created   512        131072      3840         19010 
9dd3ff89-a54a-42fb-95df-19af8d25b416 created   512        131072      32           19002 
b1af80e7-0817-4cae-974a-bd6c44c5772a created   512        131072      64           19007 
b8ab2784-68da-4d4d-ba1a-cf9aed31c0bb created   512        131072      160          19009 
bb058b0b-ffb3-4be6-b58c-601fbd7d8adf created   512        131072      2400         19008 
be80f7de-cebc-4cbc-8bd3-83d25986835a created   512        131072      160          19005 
cb7ee492-089d-4466-b16e-85f5d5c6ca3d created   4096       16384       16           19001 
root@BRM42220009:~# /tmp/omdb crucible-agent --crucible-agent-url http://\[fd00:1122:3344:107::8\]:32345 snapshots list              
REGION_ID                            SNAPSHOT_ID                          STATE   PORT  
9dd3ff89-a54a-42fb-95df-19af8d25b416 bd1a3a42-eb77-44ce-a606-49d5976db3d7 created 19003 
```

There is more to do here, but I wanted to get the first part out then iterate.

### CODE REVIEW

File: dev-tools/omdb/src/bin/omdb/crucible_agent.rs [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+use crucible_agent_client::types::RegionId;
+use crucible_agent_client::Client;
+use tabled::Tabled;
+
+use crate::Omdb;
+
+/// Arguments to the "omdb crucible-agent" subcommand
+#[derive(Debug, Args)]
+pub struct CrucibleAgentArgs {
+    /// URL of the crucible agnet internal API
```
Comment (jmpesp) [inline dev-tools/omdb/src/bin/omdb/crucible_agent.rs:20]: typo here

File: dev-tools/omdb/src/bin/omdb/crucible_agent.rs [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+
+#[derive(Debug, Subcommand)]
+enum RegionCommands {
+    /// Print list of all running control plane regions
+    List,
+}
+
+#[derive(Debug, Subcommand)]
+enum SnapshotCommands {
+    /// Print list of all running control plane regions
```
Comment (jmpesp) [inline dev-tools/omdb/src/bin/omdb/crucible_agent.rs:47]: s/regions/snapshots/

File: dev-tools/omdb/src/bin/omdb/crucible_agent.rs [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+    pub(crate) async fn run_cmd(
+        &self,
+        _omdb: &Omdb,
+    ) -> Result<(), anyhow::Error> {
+        // The crucible agent URL is required, but can come
+        // from the environment, in which case it won't be on the command line.
+        let Some(crucible_agent_url) = &self.crucible_agent_url else {
+            bail!(
+                "crucible agent URL must be specified with \
+                --crucible-agent-url or OMDB_CRUCIBLE_AGENT_URL"
```
Comment (jmpesp) [inline dev-tools/omdb/src/bin/omdb/crucible_agent.rs:62]: It's clear to me that this is referring to an environment variable, but how about changing the text to something like "or by setting the environment variable OMDB_CRUCIBLE_AGENT_URL" ?

File: dev-tools/omdb/src/bin/omdb/crucible_agent.rs [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+    extent_size: String,
+    extent_count: String,
+    port: String,
+}
+
+/// Runs `omdb crucible-agent regions list`
+async fn cmd_region_list(
+    client: &crucible_agent_client::Client,
+) -> Result<(), anyhow::Error> {
+    let regions = client.region_list().await.context("listing zones")?;
```
Comment (jmpesp) [inline dev-tools/omdb/src/bin/omdb/crucible_agent.rs:93]: s/zones/regions/

File: dev-tools/omdb/src/bin/omdb/crucible_agent.rs [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+    region_id: String,
+    snapshot_id: String,
+    state: String,
+    port: String,
+}
+/// Runs `omdb crucible-agent snapshot list`
+async fn cmd_snapshot_list(
+    client: &crucible_agent_client::Client,
+) -> Result<(), anyhow::Error> {
+    let regions = client.region_list().await.context("listing zones")?;
```
Comment (jmpesp) [inline dev-tools/omdb/src/bin/omdb/crucible_agent.rs:127]: s/zones/regions/

**Review by jmpesp — APPROVED** [review]
:rocket:

File: dev-tools/omdb/src/bin/omdb/crucible_agent.rs (hunk 1 above)
Comment (leftwo) [reply dev-tools/omdb/src/bin/omdb/crucible_agent.rs]: Fixed!

File: dev-tools/omdb/src/bin/omdb/crucible_agent.rs (hunk 2 above)
Comment (leftwo) [reply dev-tools/omdb/src/bin/omdb/crucible_agent.rs]: Fixed

File: dev-tools/omdb/src/bin/omdb/crucible_agent.rs (hunk 3 above)
Comment (leftwo) [reply dev-tools/omdb/src/bin/omdb/crucible_agent.rs]: Done

File: dev-tools/omdb/src/bin/omdb/crucible_agent.rs (hunk 4 above)
Comment (leftwo) [reply dev-tools/omdb/src/bin/omdb/crucible_agent.rs]: Done

**Pull request merged**